app/src/main/java/chat/rocket/android/authentication/signup/ui/SignupFragment.kt

```diff
@@ -19,12 +19,15 @@ import javax.inject.Inject
 
 class SignupFragment : Fragment(), SignupView {
     @Inject lateinit var presenter: SignupPresenter
-
     private val layoutListener = ViewTreeObserver.OnGlobalLayoutListener {
-        if (KeyboardHelper.isSoftKeyboardShown(constraint_layout.rootView)) {
-            text_new_user_agreement.setVisible(false)
+        if (KeyboardHelper.isSoftKeyboardShown(relative_layout.rootView)) {
+            bottom_container.setVisible(false)
         } else {
-            text_new_user_agreement.setVisible(true)
+            bottom_container.apply {
+                postDelayed({
```

Comment:
postDelayed 3ms is a very big code smell. It would be the same as post or postOnAnimation